9.0 attachment preview gs

ghost opened the pull request.

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[ADD] viewerjs 0.5.8

hbrunn committed:
[ADD] readd our tweaks

hbrunn committed:
[FIX] pass file extension where newer viewerjs versions expect it

fixes #67

hbrunn committed:
[IMP] simplify code

hbrunn committed:
[IMP] use current OCA template for README

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

Commit:
[FIX] - attachment_preview: Migrated the module to 9.0 version and fixed some errors displaying the attachments files.

Commit:
[FIX] - attachment_preview: Migrated the module to 9.0 version and fixed some errors displaying the attachments files.

Commit:
[FIX] - attachment_preview: Migrated the module to 9.0 version and fixed some errors displaying the attachments files.

Commit:
[FIX] - attachment_preview: Migrated the module to 9.0 version and fixed some errors displaying the attachments files.

pedrobaeza commented:
Please sign the CLA.

pedrobaeza commented on attachment_preview/README.rst:
```diff
@@ -9,16 +9,30 @@ Installation
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/118/8.0
+
+For further information, please visit:
```
Remove this part

pedrobaeza commented on attachment_preview/README.rst:
```diff
@@ -9,16 +9,30 @@ Installation
 
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/118/8.0
```
9.0

pedrobaeza reviewed (approved):
Little comment, but code review + try on runbot correct. Most of the changes are from the new version library.

pedrobaeza commented:
Please squash all Transbot commits together and all your commits also.

hbrunn commented on attachment_preview/static/src/css/attachment_preview.css:
```diff
@@ -1,11 +1,16 @@
-.openerp .oe_sidebar .oe_dropdown_menu li .oe-sidebar-attachment-preview
+/*.openerp .oe_sidebar .oe_dropdown_menu li .oe_sidebar_attachment_preview
```
don't comment out stuff, remove if not necessary

hbrunn commented on attachment_preview/static/src/js/attachment_preview.js:
```diff
@@ -19,41 +19,47 @@
                 .click(this.on_attachment_preview);
             this.update_preview_buttons();
-            return result;
+           // return _super.apply(self, arguments);
```
delete

hbrunn commented on attachment_preview/static/src/js/attachment_preview.js:
```diff
@@ -208,22 +211,21 @@ openerp.attachment_preview = function(instance)
-                        if(openerp.attachment_preview.can_preview(extension))
+                        var $element = self.$('.oe_binary_preview');
+                        $element.unbind( "click" );
+                       if(can_preview(extension[0]))
```
formatting

hbrunn commented on attachment_preview/static/src/js/attachment_preview.js:
```diff
@@ -208,22 +211,21 @@ openerp.attachment_preview = function(instance)
-                                        self.view.datarecord.id),
-                                    extension,
+
+                                        ('data:application/'+extension[0]+';base64,'+
```
why would you want to do this? With a few really big attachments, your browser becomes really inresponsive

hbrunn commented on attachment_preview/static/lib/ViewerJS/index.html:
```diff
@@ -39,43 +39,41 @@
-a);return b}function x(a){a=a.split("?")[0].split(".").pop();var b=q(a);b&&console.log("Found plugin by file extension from path: "+a);return b}function r(a){var b={};(a.search||"?").substr(1).split("&").forEach(function(a){a&&(a=a.split("=",2),b[decodeURIComponent(a[0])]=decodeURIComponent(a[1]))});return b}var k,p=[function(){var a="application/vnd.oasis.opendocument.text application/vnd.oasis. ...
-b="odt fodt ott odp fodp otp ods fods ots".split(" ");return{supportsMimetype:function(b){return-1!==a.indexOf(b)},supportsFileExtension:function(a){return-1!==b.indexOf(a)},path:"./ODFViewerPlugin",getClass:function(){return ODFViewerPlugin}}}(),{supportsMimetype:function(a){return"application/pdf"===a},supportsFileExtension:function(a){return"pdf"===a},path:"./PDFViewerPlugin",getClass:function ...
-c;a?(d.title||(d.title=a.replace(/^.*[\\\/]/,"")),d.documentUrl=a,b(a,function(b){b||(b=d.type?f(d.type):x(a));b?"undefined"!==String(typeof loadPlugin)?loadPlugin(b.path,function(){c=b.getClass();new Viewer(new c,d)}):(c=b.getClass(),new Viewer(new c,d)):new Viewer})):new Viewer};k=document.createElementNS(document.head.namespaceURI,"style");k.setAttribute("media","screen");k.setAttribute("type" ...
+m.webkitReques
```
if this is not needed any more, also delete the file

hbrunn commented on attachment_preview/static/src/js/viewerjs_tweaks.js:
```diff
@@ -24,10 +24,9 @@
     }
     return original_Viewer(plugin, parameters);
-}
+};
```
you don't include this file any more, so it can go away i guess?

hbrunn commented:
adding tests to have coverage at least not drop would be good

Commit:
[FIX] - attachment_preview: Fixed error in enterprise version

Added secondary css jquery selector to include the preview buttons in
community and enterprise versions.
